packages/geoip-location/src/downloadGeoIpDatabase.ts

```diff
@@ -0,0 +1,156 @@
+import crypto from 'crypto'
+import fs from 'fs'
+import { Readable } from 'stream'
+import { finished } from 'stream/promises'
+import tar from 'tar'
+import NodePath from 'path'     // use NodePath to avoid conflict with other 'path' symbols
+import { CityResponse, Reader } from 'mmdb-lib'
+
+const GEOIP_MIRROR_URL = 'https://raw.githubusercontent.com/GitSquared/node-geolite2-redist/master/redist/'
+const DB_NAME = 'GeoLite2-City'
+const TAR_SUFFFIX = '.tar.gz'
+const DB_SUFFIX = '.mmdb'
+const HASH_SUFFIX = '.mmdb.sha384'
+
+const downloadNewDb = async (url: string, dbPath: string, remoteHash: string, 
```

Comment:
Wondering what happens if permissions do not allow for writing or reading in the directory